src/main/java/os/failsafe/executor/FailsafeExecutor.java

```diff
@@ -89,13 +89,21 @@ public FailsafeExecutor(SystemClock systemClock, DataSource dataSource, int work
      * Start execution of any submitted tasks.
      */
     public void start() {
+        start(null);
+    }
+
+    /**
+     * Start execution of any submitted tasks.
+     * @param nodeId id of the node that locks the given task
+     */
+    public void start(String nodeId) {
         boolean shouldStart = running.compareAndSet(false, true);
         if (!shouldStart) {
             return;
         }
 
         executor.scheduleWithFixedDelay(
-                this::executeNextTasks,
+                () -> executeNextTasks(nodeId),
```

Comment:
instead of passing nodeId as parameter, I would assign it to a private volatile variable, that can be used within this class